src/app/dim-ui/WeaponDeepsightInfo.tsx

```diff
@@ -9,17 +16,70 @@ import styles from './WeaponDeepsightInfo.m.scss';
 export function WeaponDeepsightInfo({ item }: { item: DimItem }) {
   const deepsightInfo = item.deepsightInfo;
   const record = item.patternUnlockRecord;
+  const relevantObjectives = record?.objectives.filter((o) => !o.complete);
 
-  if (!deepsightInfo && !record?.objectives[0]) {
+  if (!deepsightInfo && !relevantObjectives?.length) {
     return null;
   }
 
   return (
     <div className={styles.deepsightProgress}>
-      {deepsightInfo && <Objective objective={deepsightInfo.attunementObjective} />}
-      {record?.objectives.map((objective) => (
-        <Objective key={objective.objectiveHash} objective={objective} />
-      ))}
+      {deepsightInfo ? (
+        <>
+          <PatternUnlockedIndicator record={record} />
+          <div className={styles.deepsightProgressSection}>
+            <Objective objective={deepsightInfo.attunementObjective} />
+          </div>
+        </>
+      ) : (
+        Boolean(relevantObjectives?.length) && (
+          <div className={styles.deepsightProgressSection}>
+            {relevantObjectives!.map((objective) => (
```

Comment:
Or the `!.` here?

Reply:
TS doesn't find the L35 boolean wrapper convincing enough on L3,
so i just changed it to the longer, more explicit non-chained check